### PR TITLE
Fix database password visibility toggle

### LINKED ISSUE #20833: [19.0] Broken dbpassword-toggle button
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

The dbpassword-toggle button is broken in current 19.0-beta. It works fine in 18.0.

### Steps to reproduce
1. fetch the sources for Nextcloud 19 on a webserver an access from a browser (Mozilla Firefox)
2. toggle the Database to something else than SQLite
3. provide any database password and toggle the eye symbol (dbpassword-toggle)

### Expected behaviour
password should be shown in clear text

### Actual behaviour
nothing happens

no logs or error are shown for this action.

### LINKED ISSUE #22470: database password not viewable during installation nextcloud 19.0.3 and 20.0.0
In the latest server versions of nextcloud the error for which the database password cannot be displayed during the installation of the nextcloud server has returned. if you click on the eyes icon to view it, nothing happens. I attach image

Immagine

### PR DESCRIPTION
**Pull request by st3iny**
Fixes #20833, Fixes https://github.com/nextcloud/server/issues/22470

### CODE REVIEW

**Comment by st3iny** [comment]
@J0WI @MarcinWieczorek @skjnldsv Just stumbled across this bug. Your reviews are appreciated.

File: core/src/install.js [hunk 1]
```diff
@@ -134,4 +136,6 @@ window.addEventListener('DOMContentLoaded', function() {
 		drawTitles: true,
 		nonce: btoa(getToken()),
 	})
+
+	$("#dbpass").showPassword().keyup()
```
Comment (juliushaertl) [inline core/src/install.js:140]: Nice catch, let's also re-add the missing toggle for the admin password then from https://github.com/nextcloud/server/commit/bba26cb2ebf48413ed938af572ee46b7cb9f0ba2#diff-e60184f15246b33f90d369d78f0f2df1L3

```suggestion
	$('#adminpass').showPassword().keyup()
	$("#dbpass").showPassword().keyup()
```

Comment (st3iny) [reply core/src/install.js:140]: The show button is not rendered correctly on the adminpass field. I will add a commit that fixes the template too. Just adding this line won't do anything.

**Comment by st3iny** [comment]
If you are happy with the changes I can squash the commits.

File: core/templates/installation.php [hunk 2]
```diff
@@ -50,7 +50,7 @@
 				autocomplete="off" autocapitalize="none" autocorrect="off" required>
 			<label for="adminpass" class="infield"><?php p($l->t('Password')); ?></label>
 			<input type="checkbox" id="show" class="hidden-visually" name="show">
-			<label for="show" class="hidden-visually"><?php p($l->t('Show password')); ?></label>
+			<label for="show"></label>
```
Comment (skjnldsv) [inline core/templates/installation.php:53]: Please keep the string, this is here for screen readers :)

Comment (st3iny) [reply core/templates/installation.php:53]: I did not know that. The icon is invisible when due to the hidden-visually class. Without it the label text will be displayed on top of the input field which does not look nice. 

I fixed it by replacing `left: -10000px` with `text-indent: -10000px` in the css-class hidden-visually. Is that okay or should I add a style property to this specific label?

Comment (skjnldsv) [reply core/templates/installation.php:53]: That is why the hidden visually is here, it hides it to the view, but not to the browser.
Instead you can also use `aria-label`, it won't show.
On the input directly, add `aria-label="<?php p($l->t('Show password')); ?>"`
It should be ok then :)

Comment (st3iny) [reply core/templates/installation.php:53]: Using `aria-label` is a great idea. I will add one to the database password field too.

**Comment by st3iny** [comment]
The strengthify meter of the admin password field has an obsolete left margin. Should we fix it in this PR too? A new PR for such a small fix may be a bit too much overhead.

strengthify-bug

**Review by skjnldsv — APPROVED** [review]

**Comment by rullzer** [comment]
So this is ready for review now I guess?

**Review by GretaD — APPROVED** [review]

**Comment by rullzer** [comment]
@skjnldsv 

```
07:42:25.037Z ERROR probot: Error during the compiling or the committing phase
```

**Comment by skjnldsv** [comment]
> ```
> 07:42:25.037Z ERROR probot: Error during the compiling or the committing phase
> ```

Debug logs?

File: core/src/install.js [hunk 3]
```diff
@@ -134,4 +136,7 @@ window.addEventListener('DOMContentLoaded', function() {
...
+	$('#adminpass').showPassword().keyup()
```
Comment (skjnldsv) [inline core/src/install.js:141]:
```suggestion
	$('#dbpass').showPassword().keyup()
	$('#adminpass').showPassword().keyup()
```

Weird, it doesn't log the lint error inline.
That is why it fails

Comment (st3iny) [reply core/src/install.js]: Oops

**Comment by skjnldsv** [comment]
No more force push until the bot is finished! :warning:

**Comment by skjnldsv** [comment]
Ah damn, @st3iny please rebase to master drop the changes and force push again :)

**Comment by skjnldsv** [comment]
/compile amend /

￼

**Comment by MorrisJobke** [comment]
> The backport to stable19 failed. Please do this backport manually.

@skjnldsv ^

**Pull request merged**